CRM/Membershipextrasimporterapi/EntityImporter/Membership.php

```diff
@@ -17,12 +17,17 @@ public function __construct($rowData, $contactId, $recurContributionId) {
   }
 
   public function import() {
-    if (empty($this->rowData['membership_external_id'])) {
-      // todo : when the line item importer is created,
-      // we need to throw an exception if this field  is not set and line_item.entity_table = "civicrm_membership"
+    $membershipExternalIdSet = !empty($this->rowData['membership_external_id']);
```

Comment:
Perhapes move this before ` if (!$membershipExternalIdSet) {` as we would not need it `$membershipExternalIdSet` if `$isMembershipLineItem` does not have value

Reply:
not sure I follow, you need me to move what exactly and before which line?